### ISSUE TITLE
gbSP won’t load games

### ISSUE THREAD

**Issue by System-Lag** [ndless-nspire/Ndless#235]
I have a nSpire CX CAS (version P-1214T) running 4.5.0.1180. I have ndless installing on my calculator just fine and can open gbsp_launcher to get to the vile navigation. However, as soon as I select a game to play the calculator immediately boots out of gbsp and to the my documents file browser. I’ve deleted and reinstalled the OS, deleted and reinstalled ndless_installer_4.5.0 and ndless_resources. Deleted and reinstalled all the gbsp programs both from the internet and from a calculator which gbsp does work on, but it all has the same problem. Is there something else I need to do to get gbsp to load the games?

**Comment by System-Lag** [comment]
Also one other thing to note is I don’t get the warning message when I launch gbsp it just goes straight to the file navigation

**Comment by System-Lag** [comment]
Also one other thing to note is I don’t get the warning message when I launch gbsp it just goes straight to the file navigation

**Comment by System-Lag** [comment]
Also one other thing to note is I don’t get the warning message when I launch gbsp it just goes straight to the file navigation

**Comment by System-Lag** [comment]
Also one other thing to note is I don’t get the warning message when I launch gbsp it just goes straight to the file navigation

**Comment by System-Lag** [comment]
Also one other thing to note is I don’t get the warning message when I launch gbsp it just goes straight to the file navigation

**Comment by System-Lag** [comment]
Also one other thing to note is I don’t get the warning message when I launch gbsp it just goes straight to the file navigation

**Comment by Vogtinator** [comment]
What do you mean by "boots out of gbsp"? Does it crash or just exit?

**Comment by System-Lag** [comment]
It just exits, so I’ll hit enter to select a game, and gbsp will immediately close out of the blue file browser screen and leave me on the white my documents screen

**Comment by Vogtinator** [comment]
It might've run out of memory. Not sure how to debug that without access to the serial output though, hm.

**Comment by System-Lag** [comment]
How much memory does it need free?

**Comment by Vogtinator** [comment]
Depends on the ROM. Some ROMs require more than the available memory to run, so won't work.

**Comment by System-Lag** [comment]
Is Pokémon Crystal that demanding though?

**Comment by Vogtinator** [comment]
Yep. ROMs over 12MiB most likely won't work.

**Comment by Christian-Moore** [comment]
For the hardware it is, yes.

**Comment by System-Lag** [comment]
So what games can be played on it? I’ve played Pokémon Emerald just fine on my previous CX

**Comment by Vogtinator** [comment]
Actually, I just gave it a try and the fallback path in GPSP for low memory seems to work. However, the game has some issues, IIRC some game breaking crashes were reported at some other place.

**Comment by System-Lag** [comment]
So could it just be I don’t have the right hardware?

**Comment by Vogtinator** [comment]
Unlikely. It's possible that some hardware difference or some unknown software incompatibility causes that.

**Comment by System-Lag** [comment]
So is my best bet just waiting until a new version of ndless comes out for 4.5.X and updating to that?

**Comment by System-Lag** [comment]
Is my best bet just waiting until a new version of ndless comes out for 4.5.X and updating to that?

**Comment by Vogtinator** [comment]
I don't think that would help, unless is an OS-specific bug, which is unlikely. Does your other calc run the same OS, version of Ndless and GPSP?

**Comment by System-Lag** [comment]
Yes, at first I just transferred the gpsp folder, rom folder, and ndless folder between the two calculators

**Comment by anoopsk** [comment]
I tried Minish Cap and mario superstar saga  on my N-1018AE calc. The screen would become white and unresponsive and freeze. I tried waiting a bit(for 20 mins), reinstalled the os. But that didn't fix the problem. Some of the other gba games work on my calc.
https://github.com/Vogtinator/gpsp/issues/6

**Comment by chenxi050402** [comment]
It don't work for some games for me on 5.2.0

**Comment by RuralAnemone** [comment]
> I tried Minish Cap and mario superstar saga on my N-1018AE calc. The screen would become white and unresponsive and freeze. I tried waiting a bit(for 20 mins), reinstalled the os. But that didn't fix the problem. Some of the other gba games work on my calc.
> [Vogtinator/gpsp#6](https://github.com/Vogtinator/gpsp/issues/6)

the same thing happened to me during specific parts of those games in Ndless 5.2.0.771
Superstar Saga hardlocks (freezes and you have to reset your calc) when it's loading a battle, on the frame when the spinning star starts to fade
The Minish Cap works fine until you get outside of Vaati's castle (the purple lava + tornadoes part), where it also hardlocks
Pokemon Emerald & Sapphire both get stuck in sleep mode (the white screen thing) whenever I try to open them, but I can just press Q to savestate+quit.
Those are the only games I've had issues with.

**Comment by Facts-and-Logic** [comment]
I have the CX II running Ndless 5.2.0.771
The Minish Cap freezing problem seems to be an emulator problem more than the actual calculator. Seems like you'll have to finish the game on PC.
https://pyra-handheld.com/boards/threads/gpsp-zelda-minish-cap-freezing.46336/
(this is all the way back from 2009)

I've never had a problem with Pokemon Emerald and I was able to finish the game without any problems.

I've used the the original build of gpsp rather than the updated build (https://github.com/ndless-nspire/Ndless/issues/256) because of its performance. The newer build is slower since games run at a lower frame rate at 288 mHz while the original build doesn't at the same clock speed. Also Super Mario Advance 4 didn't work on the newer build so that also made me switch.

**Comment by Seojun618** [comment]
Can anyone send me the link for the best emulator for gba and the smallest size from for pokemon emerald? My calculator is also facing the same problem and I've been trying to find a tutorial for it.
